### PR TITLE
Support get persistent topics or non-persistent topics for pulsar admin client

### PR DESCRIPTION
**Pull request by codelipenghui**
### Motivation

Currently, we can only get all topics by the admin client. This pr supports to get persistent topic or non-persistent topic by add `--topic-domain=persistent` or `--topic-domain=non-persistent`.

For Pulsar SQL, is should only get fetch the persistent topics when list tables. I will push the next PR to fix the pulsar SQL.

### Verifying this change

Unit test and integration test added.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (yes)

The document for the admin API is generated automatically, @Jennifer88huang please double confirm.

### CODE REVIEW

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java [hunk 1]
```diff
@@ -90,6 +120,29 @@
...
+     *  "topic://my-tenant/my-namespace/topic-2"]</code>
+     * </pre>
+     *
+     * @param namespace
+     *            Namespace name
+     *
+     * @param mode
+     *            Get topic mode
+     *            use {@link GetTopicsMode#PERSISTENT} to get persistent topics
+     *            use {@link GetTopicsMode#NON_PERSISTENT} to get non-persistent topics
```
Comment (315157973) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:139]: The non-persistent topic will always return empty, so this interface cannot get the non-Persistent topic

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/GetTopicsMode.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * under the License.
+ */
+
+package org.apache.pulsar.client.admin;
+
+/**
+ * Pulsar supports persistent topic and non-persistent topic.
+ * The GetTopicsMode is used to get persistent/non-persistent topics.
+ */
+public enum GetTopicsMode {
```
Comment (315157973) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/GetTopicsMode.java:26]: Maybe we can reuse the `TopicType` enumeration, when the type is not set, then read all. This is just a suggestion.

Comment (codelipenghui) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/GetTopicsMode.java:26]: TopicType is partitioned/non-partitioned. Maybe change to TopicDomain is better.

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java (hunk 1 above)
Comment (codelipenghui) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:139]: Why non-persistent topics will always return empty? I'm not getting the key point here, currently the list topics operation combine the persistent topics and non-persistent topic, I also add test for cover this case.

Comment (315157973) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java]: In the client, if it is a non-persistent topic, the request will complete directly
https://github.com/apache/pulsar/blob/d17eb071e663780e5ba3fd6475d088193d8c6154/pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/TopicsImpl.java#L168

**Review by 315157973 — APPROVED** [review]

**Comment by Jennifer88huang** [comment]
@codelipenghui yes, it will be generated automatically from the code file. We can double check whether it's displayed on the admin website after merging the PR and triggering the website.

File: branch_delete_except.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+for branch in `git branch`; do
```
Comment (eolivelli) [inline branch_delete_except.sh:3]: It looks like this file is not related to this patch, please remove

**Review by eolivelli — CHANGES_REQUESTED** [review]
Awesome work.

File: branch_delete_except.sh (hunk 3 above)
Comment (codelipenghui) [reply branch_delete_except.sh:3]: Oh, my mistake.

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by eolivelli** [comment]
@codelipenghui  this is a new REST API, do you really want to pick it to 2.7.2 ?
usually it is better to not add new APIs on patch releases

**Comment by codelipenghui** [comment]
@eolivelli removed 2.7.2
